internal/serialization/field_operations.go

```diff
@@ -0,0 +1,373 @@
+	return VARIABLE_SIZE
+}
+
+type TimestampWithTimezoneFieldKindOperation struct{}
+
+func (TimestampWithTimezoneFieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type ArrayOfTimestampWithTimezoneFieldKindOperation struct{}
+
+func (ArrayOfTimestampWithTimezoneFieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type CompactFieldKindOperation struct{}
+
+func (CompactFieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type ArrayOfCompactFieldKindOperation struct{}
+
+func (ArrayOfCompactFieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type NullableBooleanFieldKindOperation struct{}
+
+func (NullableBooleanFieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type ArrayOfNullableBooleanFieldKindOperation struct{}
+
+func (ArrayOfNullableBooleanFieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type NullableInt8FieldKindOperation struct{}
+
+func (NullableInt8FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type ArrayOfNullableInt8FieldKindOperation struct{}
+
+func (ArrayOfNullableInt8FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type NullableInt16FieldKindOperation struct{}
+
+func (NullableInt16FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type ArrayOfNullableInt16FieldKindOperation struct{}
+
+func (ArrayOfNullableInt16FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type NullableInt32FieldKindOperation struct{}
+
+func (NullableInt32FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type ArrayOfNullableInt32FieldKindOperation struct{}
+
+func (ArrayOfNullableInt32FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type NullableInt64FieldKindOperation struct{}
+
+func (NullableInt64FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type ArrayOfNullableInt64FieldKindOperation struct{}
+
+func (ArrayOfNullableInt64FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type NullableFloat32FieldKindOperation struct{}
+
+func (NullableFloat32FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type ArrayOfNullableFloat32FieldKindOperation struct{}
+
+func (ArrayOfNullableFloat32FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type NullableFloat64FieldKindOperation struct{}
+
+func (NullableFloat64FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type ArrayOfNullableFloat64FieldKindOperation struct{}
+
+func (ArrayOfNullableFloat64FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+func FieldOperations(fieldKind pserialization.FieldKind) FieldKindOperation {
+	switch fieldKind {
+	case pserialization.FieldKindBoolean:
+		return BooleanFieldKindOperation{}
```

Comment:
I think these have zero allocation cost right? Great choice 👍 😄

Reply:
🤔 no idea :D

Reply:
Interesting read 👀 https://dave.cheney.net/2014/03/25/the-empty-struct